projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -370,6 +370,13 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseCompone
                     }
                 });
             });
+            this.parent.verticalScrollContainer.onDataChanged.pipe(takeUntil(this.destroy$)).subscribe(() => {
```

Comment:
Will this work in-depth. If multiple descendants need to get reflowed because of a parent. This binds to a parent but will the reflow cause an onDataChanged for the children to get updated?

Reply:
Applied changes to reflow children that have percentage width as well.